Scripts/developer_scripts/licensecheck

```diff
@@ -444,6 +444,10 @@ sub parselicense {
 	$license = "GENERATED FILE";
     }
 
+    if ($licensetext =~ /This file incorporates work covered by the following copyright and permission notice/i) {
+  $license = "DERIVED WORK";
+    }
+
```

Comment:
Where is that used?

Reply:
Actually, I disagree. Even if a file is a derived work from something licensed under an MIT license, the result is a license equivalent to "(GPLv3+ AND MIT) or (commercial AND MIT)".

"DERIVED WORK" cannot be a license.

But here I think we reach the limits of a script like `licensecheck`, at least in its current version. Maybe we should check if a more recent version is published somewhere.

And, anyway, what do we use "licensecheck" for, now that we have opted for SPDX headers?

Reply:
AND means that the more restrictive wins, right?
I do not know what exactly the license check is supposed to check.  That  there is a header with  any SPDX?   Or that this header corresponds to what we write in the PkgDescription.txt and in the package_info/*/license.txt ??

Reply:
> AND means that the more restrictive wins, right?

For the more general terms of the license, yes: the GPL will "win". But actually all the requirements of both licenses must be fulfilled. The MIT license of the file (at https://github.com/CGAL/cgal/pull/5090/files#diff-f0e1e12ea5078e54628eecdc01f643035add6a510138df0bf7eeb6d03e8cfd25R14-R34) says:
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.



All users of that derived work will have to comply with the terms of the GPLv3+ AND with that mandatory advertising clause.

That is why saying "GPLv3+" only is a bit wrong, and the correct licensing designation is "GPLv3+" AND "MIT". Even users of the commercial license of CGAL, they will have to comply with the MIT license as well. That is why the final result, for the SPDX tag is "(GPL-3.0-or-later AND MIT) OR (LicenseRef-Commercial AND MIT)".

> I do not know what exactly the license check is supposed to check. That there is a header with any SPDX? Or that this header corresponds to what we write in the PkgDescription.txt and in the package_info/*/license.txt ??

Since we have switched to XPDX, I do not know exactly what we should do either.

Reply:
I updated the header but I'll keep that modification of the script and keep the whole PMP package as GPL. We'll handle that correctly once the package is split.